src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenStringConcat.cs

```diff
@@ -2104,5 +2105,1072 @@ .maxstack  5
   IL_0084:  ret
 }");
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/66827")]
+        public void ConcatTwo_ReadOnlySpan_LValue()
```

Comment:
I do not understand the name of this test. There are no lvalues being used in this test.

Reply:
I thought that lvalue is something that has address in memory, thus can be taken reference from. Renamed this suffixes to `DirectReference` when we can just pass char argument to constructor of ROS by reference and `IndirectReference` when we first need to store result in a local and then pass that local by reference.